app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -126,13 +126,25 @@ class MainActivity : AppCompatActivity() {
     }
 
     override fun onBackPressed() {
-        val currentFragment = this.supportFragmentManager.findFragmentById(R.id.frameContainer)
+
         val rootFragment = this.supportFragmentManager.findFragmentById(R.id.rootLayout)
-        if (currentFragment !is EventsFragment && rootFragment !is EventDetailsFragment) {
-            loadFragment(EventsFragment())
-            navigation.selectedItemId = navigation_events
-        } else {
+        if (rootFragment is EventDetailsFragment) {
             super.onBackPressed()
+            return
         }
+
+        val currentFragment = this.supportFragmentManager.findFragmentById(R.id.frameContainer)
+        when(currentFragment) {
+            is EventsFragment -> finish()
```

Comment:
This solution will work but the app should exit even without explicitly using this method.
I guess it is because the logic of `loadFragment` was changed. Earlier `addToBackStack` wasn't used in that method.